simplified-books-covers/src/main/java/org/nypl/simplified/books/covers/BookCoverProvider.kt

```diff
@@ -129,21 +128,15 @@ class BookCoverProvider private constructor(
   }
 
   private fun coverURIOf(entry: FeedEntry.FeedEntryOPDS): URI? {
-    val bookOpt = this.bookRegistry.book(entry.bookID)
-    if (bookOpt is Some<BookWithStatus>) {
-      val book = bookOpt.get()
-      return book.book.cover?.toURI()
-    }
-    return mapOptionToNull(entry.feedEntry.cover)
+    val bookWithStatus =
+      this.bookRegistry.bookOrNull(entry.bookID)
+    return bookWithStatus?.book?.cover?.toURI() ?: mapOptionToNull(entry.feedEntry.cover)
   }
 
   private fun thumbnailURIOf(entry: FeedEntry.FeedEntryOPDS): URI? {
-    val bookOpt = this.bookRegistry.book(entry.bookID)
-    if (bookOpt is Some<BookWithStatus>) {
-      val book = bookOpt.get()
-      return book.book.cover?.toURI()
```

Comment:
This is a local file URI that is supposed to contain a reference to a cover. However, after looking through the code this value appears to never be written to. I suspect this must be to support apps like LFA?

Either way this value always returns null.

The fix was pretty simple and doesn't change the logic much. It just falls back to loading from the feed entry URI if the local file URI is not set.

Reply:
That in itself is a bug. The borrow task is supposed to grab the book cover image at the end and save it to disk. The `Book` value is then updated to hold a non-null `File` value referencing that on-disk copy...

Reply:
Ah, I see the `BookCoverFetchTask` now. Let me look into that.

Is there a reason we want to cache the cover to disk instead of just fetching from the feed entry? Picasso has a built-in disk cache.

Reply:
So, I'd actually like to merge this fix in as-is and investigate the reason that value isn't being written separately. This is a good safe fix to make anyway and I think the underlying issue isn't critical to get into the release.

Reply:
I agree!